aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotSslUtility.java

```diff
@@ -64,42 +59,15 @@ private AWSIotSslUtility() {
      * @throws KeyManagementException when SSL context cannot be created by key
      *             manager.
      */
-    protected synchronized static SSLSocketFactory getSocketFactoryWithKeyStore(KeyStore keyStore, int portNum)
+    public static SSLSocketFactory getSocketFactoryWithKeyStore(KeyStore keyStore, int portNumber)
             throws NoSuchAlgorithmException, UnrecoverableKeyException, KeyStoreException,
-            KeyManagementException {
-        portNumber = portNum;
-        return getSocketFactoryWithKeyStore(keyStore);
-    }
-
-    /**
-     * Creates a socket factory given a keystore.
-     *
-     * @param keyStore keystore containing a certificate and private key for
-     *            used in creating a secured socket.
-     * @return a socket factory for use in creating a secured socket.
-     * @throws NoSuchAlgorithmException when TLS 1.2 is not available.
-     * @throws UnrecoverableKeyException when the private key cannot be
-     *             recovered. Ususally a bad keystore password.
-     * @throws KeyStoreException when keystore cannot be created.
-     * @throws KeyManagementException when SSL context cannot be created by key
-     *             manager.
-     */
-    public static SSLSocketFactory getSocketFactoryWithKeyStore(KeyStore keyStore)
-            throws NoSuchAlgorithmException, UnrecoverableKeyException, KeyStoreException,
-            KeyManagementException {
-
+            KeyManagementException, NoSuchProviderException {
         SSLContext context;
 
         if (portNumber == 443) {
-            try {
-                // Attempt to use Conscrypt
-                Security.addProvider(new OpenSSLProvider());
-                context = SSLContext.getInstance(TLS_V_1_2, "Conscrypt");
-            } catch (NoSuchProviderException e) {
```

Comment:
Why was this catch removed?

Reply:
It was instead added to the method signature since the default provider will not work with on port 443 as service mandates alpn flag for 443